5.2.Expense Tracking Frontend Implementation (Create Budget)

RynniaRyan opened the pull request:
I included a calendar view for a more user friendly way of selecting budget period. As well as the option to select budget type (Individual or Family).  **‼️Note: New library added use `npm install react-native-calendar` **

# Preview
<img width="170" alt="image" src="https://github.com/user-attachments/assets/0451ace9-3efe-4e64-b42f-52cd92bc7008" /><img width="169" alt="image" src="https://github.com/user-attachments/assets/9e0bc48c-cf8e-417c-952f-af31e974cf25" />



However same issue occurs as discussed ([here)](https://trello.com/c/A57L9CgB/54-53-expense-tracking-frontend-implementation-view-budgets) and user_id cannot be fetch 😢 . 

## Edit
OK Look! apparently it works on the web version but on mobile its not. I think it is the same for the other branch too becasue I was only testing with mobile up to now:
<img width="169" alt="image" src="https://github.com/user-attachments/assets/39c7f300-b67e-44f2-8a07-e6cc8887a91f" />

RynniaRyan committed:
Implemented Calendar Pop-Up

This Calendar will be used as a date range selector instead of the user manually typing in the date range.

Still have re-arranging of layout and themes to do :D

RynniaRyan committed:
Implemented Calendar Pop-Up

This Calendar will be used as a date range selector instead of the user manually typing in the date range.

Still have re-arranging of layout and themes to do :D

RynniaRyan committed:
Added Selectve Tags for Budget Type

I added the filter tag component here, in order for users to select which budget type they are creating with 'Individual' option being the default

Though i cannot test if it works due to the same related error of connecting to backend data

ref: https://github.com/Chroma-Corps/FFM-Application/pull/18#issue-2867489024

RynniaRyan committed:
Added Selectve Tags for Budget Type

I added the filter tag component here, in order for users to select which budget type they are creating with 'Individual' option being the default

Though i cannot test if it works due to the same related error of connecting to backend data

ref: https://github.com/Chroma-Corps/FFM-Application/pull/18#issue-2867489024

RynniaRyan commented:
![](https://github.trello.services/images/mini-trello-icon.png) [5.2. Expense Tracking Frontend Implementation (Create Budget)](https://trello.com/c/BzjmZiLY/37-52-expense-tracking-frontend-implementation-create-budget)

RynniaRyan committed:
Merge branch '5.2.Expense-Tracking-Frontend-Implementation-(CreateBudget)' of https://github.com/Chroma-Corps/FFM-Application into 5.2.Expense-Tracking-Frontend-Implementation-(CreateBudget)

RynniaRyan committed:
Added PeriodSelection Pop-up, Bank Horizontal Scroll, Budget Type Selection, & Re-arranged Card Layout

I am having issues with the calendar pop up he just does not want to show at all.

working with dummydata for bank (did not test create button yet)

Still need to figure out how to make all date selection stuff work together as that is a bit confusing.  So still in Progress

RynniaRyan committed:
Edited Period Selection Popup & Calendar Popup

The only thing that is left to edit is the logic of date durations which is a bit tricky for me.

Bank Card display still comes from dummy data as the backend is not yet populated, and Category options remains empty for the same reason (for now).

RynniaRyan committed:
Installing Packages

RynniaRyan commented:
### **Pull Request Update!** 
I made changes to better align with the wireframes I created and included accommodations based on prior review.

### Pending Fixes:
- Calculation of start and end date
- Improved wording for the selected period, e.g. 'month(s)' when 'Monthly' is chosen.
- Scrollable Category view (Horizontal)
- Bank Card Preview Info

## Screen Preview
<img width="168" alt="image" src="https://github.com/user-attachments/assets/ea1d8f40-8ae0-4939-a0ff-48b64404d899" />

JaleneA reviewed (changes requested):
![image](https://github.com/user-attachments/assets/bb533010-a2a1-4607-84c7-aa129d341520)
I know I already informed you of this on the group chat, just doing it here to be official, the mobile view of the create budget screen looks like this

JaleneA committed:
UI Fix (Mobile)

Fixed the UI of the create budgets screen as it had a layout issue, further made simple design edits to different areas. Code still needs to implement proper period selection before merging

RynniaRyan commented:
@JaleneA Thanks so much for fixing that display error on mobile for me ✨ 

### **Pull Request Update!** 
### Fixed:
- Calculation of start and end date (finally 🎉 )
- Improved wording for the selected period, e.g. 'month(s)' when 'Monthly' is chosen.

Also Can we leave the category selection empty for now? since its not tied within models yet?

## Screen Preview
<img width="170" alt="image" src="https://github.com/user-attachments/assets/ea5b9d1e-4bb4-4fca-9057-b8809aeef03d" />
<img width="170" alt="image" src="https://github.com/user-attachments/assets/b094b1d5-4c61-4e81-b010-08cece55b8f0" />
<img width="170" alt="image" src="https://github.com/user-attachments/assets/a9f74158-1c0a-4881-b796-2281e2630edc" />

RynniaRyan committed:
Fixed Calculation of Budget Period

RynniaRyan committed:
Added Default Category Scroll & Enabled Selectable Bank Cards

RynniaRyan commented:
### **Pull Request Update!** 
**Ok! final fix for now until  mid-semester presentations are finished**
@JaleneA @TamiaWill 

### Edits:
- Created scrollable category view with dummy data (horizontal)
- Displayed bank card preview info
- Enabled selectable bank cards for budget creation

<img width="156" alt="image" src="https://github.com/user-attachments/assets/528e5f65-ef13-46d9-99ef-1698a87b2631" />

JaleneA reviewed (approved):
Let's great! Nice work ✨

JaleneA committed:
Merge branch '5.Expense-Tracking-Implementation' into 5.2.Expense-Tracking-Frontend-Implementation-(CreateBudget)
